### PR TITLE
feat(): ECR service and checks

### PR DESCRIPTION
**Pull request by n4ch04**
### Context

It is needed to migrate ECR checks to prowler v3

### Description

Includes ECR checks and service 

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: providers/aws/services/ecr/ecr_repositories_not_publicly_accessible/ecr_repositories_not_publicly_accessible.metadata.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    "Severity": "critical",
+    "ResourceType": "AwsEcrRepository",
+    "Description": "Ensure there are no ECR repositories set as Public",
+    "Risk": "A repository policy that allows anonymous access may allow anonymous users to perform actions.",
+    "RelatedUrl": "",
+    "Remediation": {
+      "Code": {
+        "CLI": "",
+        "NativeIaC": "https://docs.bridgecrew.io/docs/public_1-ecr-repositories-not-public#cloudformation",
+        "Other": "",
```
Comment (MrCloudSec) [inline providers/aws/services/ecr/ecr_repositories_not_publicly_accessible/ecr_repositories_not_publicly_accessible.metadata.json]:
```suggestion
        "Other": "https://docs.bridgecrew.io/docs/public_1-ecr-repositories-not-public#aws-console",
```

File: providers/aws/services/ecr/ecr_repositories_not_publicly_accessible/ecr_repositories_not_publicly_accessible.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            report.resource_id = repository.name
+            report.resource_arn = repository.arn
+            report.status = "PASS"
+            report.status_extended = f"Repository {repository.name} is not open"
+            if repository.policy:
+                for statement in repository.policy["Statement"]:
+                    if (
+                        statement["Effect"] == "Allow"
+                        and "AWS" in statement["Principal"]
+                        and statement["Principal"]["AWS"] == "*"
```
Comment (MrCloudSec) [inline providers/aws/services/ecr/ecr_repositories_not_publicly_accessible/ecr_repositories_not_publicly_accessible.py]: What about Principal:* ?

File: providers/aws/services/ecr/ecr_repositories_scan_images_on_push_enabled/ecr_repositories_scan_images_on_push_enabled.metadata.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+    "Provider": "aws",
+    "CheckID": "ecr_repositories_scan_images_on_push_enabled",
+    "CheckTitle": "Check if ECR image scan on push is enabled",
+    "CheckType": ["Identify", "Vulnerability, patch, and version management"],
```
Comment (sergargar) [inline providers/aws/services/ecr/ecr_repositories_scan_images_on_push_enabled/ecr_repositories_scan_images_on_push_enabled.metadata.json:5]:
```suggestion
    "CheckType": ["Identify", "Vulnerability patch and version management"],
```

File: providers/aws/services/ecr/ecr_repositories_scan_images_on_push_enabled/ecr_repositories_scan_images_on_push_enabled.metadata.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "SubServiceName": "",
+    "ResourceIdTemplate": "arn:partition:service:region:account-id:resource-id",
+    "Severity": "medium",
+    "ResourceType": "AwsEcrRepository",
+    "Description": "Check if ECR image scan on push is enabled",
+    "Risk": "Amazon ECR image scanning helps in identifying software vulnerabilities in your container images. Amazon ECR uses the Common Vulnerabilities and Exposures (CVEs) database from the open-source Clair project and provides a list of scan findings. ",
+    "RelatedUrl": "",
+    "Remediation": {
+      "Code": {
+        "CLI": "aws ecr create-repository --repository-name name --image-scanning-configuration scanOnPush=true--region us-east-2",
```
Comment (MrCloudSec) [inline providers/aws/services/ecr/ecr_repositories_scan_images_on_push_enabled/ecr_repositories_scan_images_on_push_enabled.metadata.json]:
```suggestion
        "CLI": "aws ecr create-repository --repository-name <repo_name> --image-scanning-configuration scanOnPush=true--region <region_name>",
```

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Great job 🔝 Please, check my comments.

File: providers/aws/services/ecr/ecr_repositories_not_publicly_accessible/ecr_repositories_not_publicly_accessible.py (hunk 2 above)
Comment (MrCloudSec) [reply providers/aws/services/ecr/ecr_repositories_not_publicly_accessible/ecr_repositories_not_publicly_accessible.py]: Solved

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**